ecommerce/api_test.py

```diff
@@ -719,3 +734,92 @@ def test_not_redeemable(self):
             assert pick_coupons(self.user) == []
         for coupon in Coupon.objects.all().exclude(id=self.not_auto_or_attached_coupon.id):
             _is_coupon_redeemable.assert_any_call(coupon, self.user)
+
+
+class PriceTests(MockedESTestCase):
+    """
+    Tests for calculating prices
+    """
+
+    def test_calculate_run_price_no_coupons(self):
+        """
+        If there are no coupons for this program the price should be what get_formatted_course_price returned
+        """
+        course_run, user = create_purchasable_course_run()
+        # This coupon is for a different program
+        coupon = CouponFactory.create()
+        UserCoupon.objects.create(coupon=coupon, user=user)
+        discounted_price = 5
+        fa_price = get_formatted_course_price(course_run.course.program.programenrollment_set.first())['price']
+        with patch('ecommerce.api.calculate_coupon_price', autospec=True) as _calculate_coupon_price:
```

Comment:
we're already asserting below that `calculate_run_price(course_run, user) == (fa_price, None)`, effectively asserting that no coupon was found/used. i don't see the value in patching this function and asserting that it wasn't called